torch/distributed/_tensor/device_mesh.py

```diff
@@ -269,3 +272,68 @@ def get_coordinate(self) -> Optional[List[int]]:
         dimensions of the mesh. If this rank is not part of the mesh, return None.
         """
         return self._coordinate_on_dim if self._coordinate_on_dim else None
+
+
+@no_type_check
+def init_device_mesh(
+    device_type: str,
+    mesh_dims: Optional[Sequence[int]] = None,
+    mesh_dim_names: Optional[Sequence[str]] = None,
+) -> DeviceMesh:
+    """
+    Initializes a `DeviceMesh` based on the device_type. If no mesh_dims and mesh_dim_names
+    are provided, a `DeviceMesh` with a single mesh dimension equal to the world size is initialized.
+    If mesh_dims and mesh_dim_names are provided, it will create a `DeviceMesh` with n-d dimensional
+    array as specified in `mesh_dims`, with each mesh_dim named by a string in `mesh_dim_names`
+    respectively.
+
+    Args:
+        device_type (str): device type of the mesh. Currently supports: cpu, cuda/cuda-like.
+
+    Kwargs:
+        mesh_dims: Optional[Sequence[int]]: A sequence describes the dimension of the multi-dimesnion array
+        that describes the layout of devices. If not specified, it defaults to `[dist.get_world_size()]`.
+        mesh_dim_names: Optional[Sequence[str]]: A sequence of mesh dim names to be assigned to each dimension
+        of the multi-dimensional array that describes the layout of devices. Its length must match the length
+        of `mesh_dims`.
+
+    Returns:
+        A :class:`DeviceMesh` object
+
+
+    Example:
+        >>> # xdoctest: +SKIP
+        >>> default_mesh = init_device_mesh("cuda")
+        >>> two_d_mesh = init_device_mesh("cuda", mesh_dims=[2, 8], mesh_dim_names=["dp", "tp"])
+        >>> two_d_mesh = init_device_mesh("cuda", mesh_dims=[2, -1], mesh_dim_names=["dp", "tp"])
+
+        >>> model_state_dict = my_model.state_dict()
+
+        >>> fs_storage_writer = torch.distributed.checkpoint.FileSystemWriter("/checkpoint/1")
+        >>> torch.distributed.checkpoint.save_state_dict(
+        >>>     state_dict=model_state_dict,
+        >>>     storage_writer=fs_storage_writer,
+        >>> )
+    """
+    if mesh_dims is None and mesh_dim_names is None:
+        return DeviceMesh(
+            device_type=device_type, mesh=torch.arange(dist.get_world_size())
```

Comment:
This looks wrong to me. I think we should make the `mesh_dims/shape` to be non-optional. Because here we are calling `dist.get_world_size()` but this call requires a default pg to be set up already. But we want to make sure user only calls `init_device_mesh` instead of first requiring them initialize a world pg.

Another option is to keep the mesh_dims be optional but we need to check the envvar to see whether there's a world_size field, and use that to create the mesh tensor

Reply:
Makes sense. Agree. If we want init_device_mesh to be the alternative for  init_process_group, then maybe we just remove this default option.